services/src/main/java/org/keycloak/services/Urls.java

```diff
@@ -273,4 +277,33 @@ private static UriBuilder themeBase(URI baseUri) {
     public static URI samlRequestEndpoint(final URI baseUri, final String realmName) {
         return realmBase(baseUri).path(RealmsResource.class, "getProtocol").build(realmName, SamlProtocol.LOGIN_PROTOCOL);
     }
+
+    public static void checkUrl(KeycloakSession session, String url, String name) throws IllegalArgumentException{
```

Comment:
I don't think this is the correct place to add validation for Urls. How about adding it to org.keycloak.validation.UrlValidation instead?

Reply:
What if we move to `org.keycloak.common.util.UriUtils` instead of creating a new class?